topics/dev/tutorials/interactive-tools/tutorial.md

```diff
@@ -0,0 +1,872 @@
+# The application
+
+The application that we will wrap in this tutorial is a simple web tool which
+allows the user to upload `csv` and `tsv` files, manipulate them and download
+them. Our application is based on an R Shiny App hosted with Shiny server.
+
+Note that there is no link between this Interactive Tool and the Galaxy history.
+More complex applications might be able to read and write outputs to the user's
+history to create a more integrated experience - see the
+[Additional components section](#galaxy-history-interaction)
+to see an example of how this can be done.
+
+Our example can already be found [online](https://github.com/Lain-inrae/geoc-gxit).
+In the following sections, we will study how it was built.
+
+
+> ### {% icon hands_on %} Hands-on
+>
+> First, let's clone the repository to take a quick look at it.
+>
+> ```console
+> $ git clone https://github.com/Lain-inrae/geoc-gxit
+> $ cd geoc-gxit
+>
+> $ tree .
+> ├── Dockerfile
+> ├── interactivetool_tabulator.xml
+> ├── gxit
+> │   ├── app.R
+> │   └── install.R
+> ├── Makefile
+> └── README.md
+> ```
+> You'll find a Galaxy tool XML, a Dockerfile and two R scripts that will be injected into the container image.
+>
+{: .hands_on}
+
+## The R scripts
+
+`app.R` defines the R Shiny application.
+
+`install.R` will be used by the docker container to install the R packages needed to run `app.R`.
+
+
+## The Dockerfile
+
+> ### {% icon tip %} A brief primer on Docker
+> Docker allows an entire application context to be containerized. A typical web application consists of an operating system, installed dependancies, web server configuration, database configuration and, of course, the codebase of the software itself. A Docker container can encapsulate all of these components in a single "image", which can be run on any machine with Docker installed.
+>
+> **Essentials of Docker:**
+>
+> 1. Write an image recipe as a Dockerfile. This single file selects an OS, installs software, pulls code repositories and copies files from the host machine (your computer).
+> 2. Build the image from your recipe:
+>
+>    `sudo docker build -t <image_name> .`
+> 3. View existing images with
+>
+>    `sudo docker image list`
+> 4. Run a container with a specified command:
+>
+>    `sudo docker run <image_name> <command>`
+> 5. View running containers:
+>
+>    `sudo docker ps`
+> 6. Stop a running container:
+>
+>    `sudo docker stop <container_name>`
+> 7. Remove a stopped container:
+>
+>    `sudo docker container rm <container_name>`
+> 8. Remove an image:
+>
+>    `sudo docker image rm <container_name>`
+{: .tip}
+
+
+Let's check out
+[the Dockerfile](https://github.com/Lain-inrae/geoc-gxit/blob/master/Dockerfile)
+that we'll use to containerize our application.
+
+This container recipe can be used to build a Docker image which can be pushed to a
+container registry in the cloud, ready for consumption by our Galaxy instance:
+
+```dockerfile
+# build on an existing container
+FROM debian:testing
```

Comment:
Any reason to not build this on rocker/r-shiny?

Reply:
I have rebuilt this to be simpler - just have to PR against Lain's repo and add to the tutorial

Reply:
Ok, that works! Thanks @neoformit